File: DSA/VaniAgrawal_590015846/Day_52_Question_1.py
# Implement two-pointer convergence on sorted arrays to efficiently detect pairs with target sums using optimal time complexity.
arr = list(map(int, input("Enter array elements separated by spaces: ").split()))
x = int(input("Enter target sum x: "))

arr.sort()

left = 0
right = len(arr) - 1
found = False

while left < right:
    current_sum = arr[left] + arr[right]
    if current_sum == x:
        found = True
        break
    elif current_sum < x:
        left += 1
    else:
        right -= 1

print("true" if found else "false")

File: DSA/Ayush_Mishra_590014844/Day_52_Question_1.c
#include <stdio.h>

int hasPairWithSum(int arr[], int n, int x) {
    int left = 0, right = n - 1;
    while (left < right) {
        int sum = arr[left] + arr[right];
        if (sum == x)
            return 1;
        else if (sum < x)
            left++;
        else
            right--;
    }
    return 0;
}

int main() {
    int n, x;
    printf("Enter size of sorted array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter sorted array elements: ");
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    printf("Enter target sum: ");
    scanf("%d", &x);
    if (hasPairWithSum(arr, n, x))
        printf("true\n");
    else
        printf("false\n");
    return 0;
}
